oeps/oep-0050-hooks-extension-framework.rst

```diff
@@ -0,0 +1,227 @@
+===================================
+OEP-0050: Hooks extension framework
+===================================
+
+.. list-table::
+   :widths: 25 75
+
+   * - OEP
+     - :doc:`OEP-0050 </oeps/oep-0050-arch-hook-based-extensions>`
+   * - Title
+     - Open edX Hooks Extension Framework
+   * - Last Modified
+     - 2021-02-04
+   * - Authors
+     - Felipe Montoya <felipe.montoya@edunext.co>
+   * - Arbiter
+     - < TBD >
+   * - Status
+     - Draft
+   * - Type
+     - Architecture
+   * - Created
+     - 2021-02-04
+   * - Review Period
+     - < TBD >
+
+Abstract
+========
+
+To sustain the growth of the Open edX ecosystem, the business rules of the
+platform must be open for extension following the open-closed principle. This
+OEP specifies a solution using what is commonly referred to as hooking. That is
+the definition of a stable set of places in the code (triggers) where externally
+defined actions (via plugins) can take place.
```

Comment:
Later in the document, the concept of _filters_ is introduced, but when referring to these external functions that can extend the functionality, only _actions_ are mentioned. That made me think that filters are also actions, if so, it should be indicated that they are. If not, where only _actions_ are mentioned, _or filters_ should be added if appropriate.

Reply:
I clarified by better defining actions and filters in the motivation and rationale sections.